cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -692,7 +693,12 @@ pub mod pallet {
 				vfp.relay_parent_number,
 			));
 
-			Ok(PostDispatchInfo { actual_weight: Some(total_weight), pays_fee: Pays::No })
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
+				total_weight,
+				DispatchClass::Mandatory,
+			);
+
+			Ok(Pays::No.into())
```

Comment:
Does it make a difference here if we just return this vs `Ok(PostDispatchInfo { actual_weight: Some(total_weight), pays_fee: Pays::No })`? It would look more correct to me

Reply:
That is what we were doing before (this PR removes this line) and it is incorrect. This is because you cant (or should not) return a post dispatch weight that is higher than the pre dispatch weight. It will not be charged. Hence the error that was emitted by try runtime.